lib/postgresql.js

```diff
@@ -354,34 +354,41 @@ PostgreSQL.prototype.applyPagination = function(model, stmt, filter) {
   return stmt.merge(limitClause);
 };
 
-PostgreSQL.prototype.buildExpression =
-  function(columnName, operator, columnValue, propertyValue) {
-    if (propertyValue instanceof RegExp) {
-      columnValue = "'" + propertyValue.source + "'";
-      if (propertyValue.ignoreCase) {
-        return new ParameterizedSQL(columnName + ' ~* ?', [columnValue]);
-      } else {
-        return new ParameterizedSQL(columnName + ' ~ ?', [columnValue]);
-      }
-    }
-    switch(operator) {
-      case 'like':
-        return new ParameterizedSQL({
-          sql: columnName + " LIKE ? ESCAPE '\\'",
-          params: [columnValue]
-        });
-      case 'nlike':
-        return new ParameterizedSQL({
-          sql: columnName + " NOT LIKE ? ESCAPE '\\'",
-          params: [columnValue]
-        });
-      default:
-        // Invoke the base implementation of `buildExpression`
-        var exp = this.invokeSuper('buildExpression',
-          columnName, operator, columnValue, propertyValue);
-        return exp;
+PostgreSQL.prototype.buildExpression = function(columnName, operator,
+    columnValue, propertyValue) {
+  if (operator === 'regexp' && typeof columnValue === 'string') {
```

Comment:
Is there any reason that you don't want make this block part of the switch/case?